classes/Mail.php

```diff
@@ -869,6 +869,13 @@ public static function mimeEncode($string, $charset = 'UTF-8', $newline = "\r\n"
      * @param string $to Email address
      *
      * @return string
+     *
+     * Try to use INTL_IDNA_VARIANT_UTS46 only if defined or PHP version >= 8.0.0
+     * If not, use INTL_IDNA_VARIANT_2003
+     * PHP 7.2.0 up to 8.0.0 will return E_DEPRECATED notices but INTL_IDNA_VARIANT_2003 will still work.
+     * From PHP 8.0.0, we cannot longer use INTL_IDNA_VARIANT_2003 so we'll get an Exception using
+     * default if INTL_IDNA_VARIANT_UTS46 not defined.
+     * See https://wiki.php.net/rfc/deprecate-and-remove-intl_idna_variant_2003
```

Comment:
Please move this block before the `@param` line to comply with Phpdoc syntax.

I think it also could be shortened just to explain that it tries using `INTL_IDNA_VARIANT_UTS46` if defined, then try `INTL_IDNA_VARIANT_2003`.

Reply:
@eternoendless I've cut the explanation shorter
@PierreRambaud Thanks!